File: hsg12-tinh/2025-2026-lamdong/cambay/cambay.py
import os
import sys

input_file = os.path.join(sys.path[0], "cambay.inp")
output_file = os.path.join(sys.path[0], "cambay.out")

n, m = 0, 0
matrix = []

min_value = float('inf')
max_value = float('-inf')

result = 0


def input():
    global n, m, matrix, min_value, max_value, result

    sys.stdin = open(input_file, 'r')

    # Đọc toàn bộ dữ liệu vào bộ nhớ đệm và tách thành các token
    input_data = sys.stdin.read().split()

    iterator = iter(input_data)

    n = int(next(iterator))
    m = int(next(iterator))

    total_elements = n * m
    matrix = [0] * total_elements

    # Đọc dữ liệu và chuyển đổi luôn sang mảng một chiều
    for i in range(total_elements):
        value = int(next(iterator))
        matrix[i] = value
        
        # Tìm giá trị nhỏ nhất
        if value < min_value:
            min_value = value

        # Tìm giá trị lớn nhất
        if value > max_value:
            max_value = value
        
    
def process():
    global n, m , matrix, result

    trap_rows = [False] * n
    trap_cols = [False] * m

    for r in range(n):
        for c in range(m):
            i = r * m + c

            # Lấy ra giá trị tại hàng r, cột c
            value = matrix[i]

            if value == min_value or value == max_value:            
                # Nếu giá trị lấy ra bằng nhỏ nhất hoặc lớn nhất
                # thì đánh dấu hàng r, cột c có bẫy
                trap_rows[r] = True
                trap_cols[c] = True

    # Đếm số hàng không có bẫy
    no_trap_row_count = trap_rows.count(False)

    # Đếm số cột không có bẫy
    no_trap_col_count = trap_cols.count(False)

    result = no_trap_row_count * no_trap_col_count


def output():
    global result

    sys.stdout = open(output_file, 'w')
    print(result)


if __name__ == '__main__':
    input()
    process()
    output()

File: hsg12-tinh/2025-2026-lamdong/cambay/cambay.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define input_file "cambay.inp"
#define output_file "cambay.out"

using namespace std;

const int INF = 2e9;

int n, m;
vector<int> matrix;

int min_value = INF;
int max_value = -1;

int result;

void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    freopen(input_file, "r", stdin);

    cin >> n >> m;

    matrix.resize(n * m, 0);

    // Đọc dữ liệu và chuyển đổi luôn sang mảng một chiều
    for (int r = 0; r < n; ++r)
    {
        for (int c = 0; c < m; ++c)
        {
            int i = r * m + c;
            cin >> matrix[i];

            // Tìm giá trị nhỏ nhất
            if (matrix[i] < min_value)
                min_value = matrix[i];

            // Tìm giá trị lớn nhất
            if (matrix[i] > max_value)
                max_value = matrix[i];
        }
    }
}

void process()
{
    vector<bool> trap_rows(n, false);
    vector<bool> trap_cols(m, false);

    int value;

    for (int r = 0; r < n; ++r)
    {
        for (int c = 0; c < m; ++c)
        {
            // Lấy ra giá trị tại hàng r, cột c
            value = matrix[r * m + c];

            if (value == min_value || value == max_value)
            {
                // Nếu giá trị lấy ra bằng nhỏ nhất hoặc lớn nhất
                // thì đánh dấu hàng r, cột c có bẫy
                trap_rows[r] = true;
                trap_cols[c] = true;
            }
        }
    }

    // Đếm số hàng không có bẫy
    int no_trap_row_count = count(trap_rows.begin(), trap_rows.end(), false);

    // Đếm số cột không có bẫy
    int no_trap_col_count = count(trap_cols.begin(), trap_cols.end(), false);

    result = no_trap_row_count * no_trap_col_count;
}

void output()
{
    freopen(output_file, "w", stdout);

    cout << result;
}

int main()
{
    input();
    process();
    output();

    return 0;
}